File: src/main/java/com/ctre/phoenix/platform/Platform.java
/*
 *  Software License Agreement
 *
 * Copyright (C) Cross The Road Electronics.  All rights
 * reserved.
 * 
 * Cross The Road Electronics (CTRE) licenses to you the right to 
 * use, publish, and distribute copies of CRF (Cross The Road) firmware files (*.crf) and Software
 * API Libraries ONLY when in use with Cross The Road Electronics hardware products.
 * 
 * THE SOFTWARE AND DOCUMENTATION ARE PROVIDED "AS IS" WITHOUT
 * WARRANTY OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
 * LIMITATION, ANY WARRANTY OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE, TITLE AND NON-INFRINGEMENT. IN NO EVENT SHALL
 * CROSS THE ROAD ELECTRONICS BE LIABLE FOR ANY INCIDENTAL, SPECIAL, 
 * INDIRECT OR CONSEQUENTIAL DAMAGES, LOST PROFITS OR LOST DATA, COST OF
 * PROCUREMENT OF SUBSTITUTE GOODS, TECHNOLOGY OR SERVICES, ANY CLAIMS
 * BY THIRD PARTIES (INCLUDING BUT NOT LIMITED TO ANY DEFENSE
 * THEREOF), ANY CLAIMS FOR INDEMNITY OR CONTRIBUTION, OR OTHER
 * SIMILAR COSTS, WHETHER ASSERTED ON THE BASIS OF CONTRACT, TORT
 * (INCLUDING NEGLIGENCE), BREACH OF WARRANTY, OR OTHERWISE
 */
package com.ctre.phoenix.platform;

public class Platform {
    
    public static int simCreate(DeviceType type, int id) {
        return PlatformJNI.JNI_SimCreate(type.value, id);
    }
    
    public static void simDestroy(DeviceType type, int id) {
        PlatformJNI.JNI_SimDestroy(type.value, id);
    }
    
    public static void simDestroyAll() {
        PlatformJNI.JNI_SimDestroyAll();
    }
   
    public static int simConfigGet(DeviceType type, int param, int valueToSend, int ordinal, int id) {
        return PlatformJNI.JNI_SimConfigGet(type.value, param, valueToSend, ordinal, id);
    }

    public static int simConfigSet(DeviceType type, int param, int value, int subValue, int ordinal, int id) {
        return PlatformJNI.JNI_SimConfigSet(type.value, param, value, subValue, ordinal, id);
    }

}

File: src/main/native/cpp/platform/Platform.cpp
#include "ctre/phoenix/platform/PlatformSim.h"

namespace ctre {
namespace phoenix {
namespace platform {

int32_t PlatformSim::SimCreate(DeviceType type, int id) {
    return c_SimCreate(type, id);
}
int32_t PlatformSim::SimDestroy(DeviceType type, int id) {
    return c_SimDestroy(type, id);
}
int32_t PlatformSim::SimDestroyAll() {
    return c_SimDestroyAll();
}
int32_t PlatformSim::SimConfigGet(DeviceType type, uint32_t param, uint32_t valueToSend, uint32_t & outValueReceived, uint32_t & outSubvalue, uint32_t ordinal, int32_t id) {
    return c_SimConfigGet(type, param, valueToSend, outValueReceived, outSubvalue, ordinal, id); 
}
int32_t PlatformSim::SimConfigSet(DeviceType type, uint32_t param, uint32_t value, uint32_t subValue, uint32_t ordinal, int32_t id) {
    return c_SimConfigSet(type, param, value, subValue, ordinal, id);
}
}
}
}
